src/time_series_metadata_store.jl

```diff
@@ -1,8 +1,26 @@
+const ASSOCIATIONS_TABLE_NAME = "time_series_associations"
 const METADATA_TABLE_NAME = "time_series_metadata"
 const DB_FILENAME = "time_series_metadata.db"
 
+@kwdef struct HasMetadataQueryKey
+    has_name::Bool
+    num_possible_types::Int
+    has_resolution::Bool
+    has_features::Bool
+    feature_filter::Union{Nothing, String} = nothing
+end
+
 mutable struct TimeSeriesMetadataStore
     db::SQLite.DB
+    # Caching compiled SQL statements saves 3-4 us per query query.
+    # DBInterface.jl does something similar with @prepare.
+    # We need this to be tied to our connection.
+    cached_statements::Dict{String, SQLite.Stmt}
+    # This caching allows the code to skip some string interpolations.
+    # It is experimental for PowerSimulations, which calls has_metadata frequently.
+    # It may not be necessary. Savings are less than 1 us.
+    has_metadata_statements::Dict{HasMetadataQueryKey, SQLite.Stmt}
```

Comment:
This cached applies only to the `has_time_series` path, which is the most important to optimize for PSI. And is possibly over-engineered.